synapse/handlers/federation.py

```diff
@@ -243,10 +259,6 @@ async def _maybe_backfill_inner(
             insertion_events_to_be_backfilled,
         )
 
-        if not backwards_extremities and not insertion_events_to_be_backfilled:
-            logger.debug("Not backfilling as no extremeties found.")
-            return False
-
         # we now have a list of potential places to backpaginate from. We prefer to
         # start with the most recent (ie, max depth), so let's sort the list.
         sorted_backfill_points: List[_BackfillPoint] = sorted(
```

Comment:
we could maybe use [`heapq.merge`](https://docs.python.org/3/library/heapq.html#heapq.merge) if both these lists are sorted?

Reply:
@squahtx mentioned this in https://github.com/matrix-org/synapse/pull/13635#discussion_r959916546 when I asked about any potential two already sorted lists optimizations. Do we want to do this though?

In this PR we're also adding the `limit` so we're only sorting 100 events total, I am leaning towards keeping it simple 🤷

Reply:
:shrug: